kubechain/internal/controller/taskrun/taskrun_controller_test.go

```diff
@@ -217,7 +217,93 @@ var _ = Describe("TaskRun Controller", func() {
 		})
 	})
 	Context("ReadyForLLM -> Error", func() {
-		XIt("moves to Error if the LLM returns an error", func() {})
+		It("moves to Error state but not Failed phase on general error", func() {
+			_, _, _, _, teardown := setupSuiteObjects(ctx)
+			defer teardown()
+
+			taskRun := testTaskRun.SetupWithStatus(ctx, kubechain.TaskRunStatus{
+				Phase: kubechain.TaskRunPhaseReadyForLLM,
+				ContextWindow: []kubechain.Message{
+					{
+						Role:    "system",
+						Content: testAgent.system,
+					},
+					{
+						Role:    "user",
+						Content: testTask.message,
+					},
+				},
+			})
+			defer testTaskRun.Teardown(ctx)
+
+			By("reconciling the taskrun with a mock LLM client that returns an error")
+			reconciler, recorder := reconciler()
+			mockLLMClient := &llmclient.MockRawOpenAIClient{
+				Error: fmt.Errorf("connection timeout"),
+			}
+			reconciler.newLLMClient = func(apiKey string) (llmclient.OpenAIClient, error) {
+				return mockLLMClient, nil
+			}
+
+			_, err := reconciler.Reconcile(ctx, reconcile.Request{
+				NamespacedName: types.NamespacedName{Name: testTaskRun.name, Namespace: "default"},
+			})
+			Expect(err).To(HaveOccurred())
+
+			By("checking the taskrun status")
+			Expect(k8sClient.Get(ctx, types.NamespacedName{Name: testTaskRun.name, Namespace: "default"}, taskRun)).To(Succeed())
+			Expect(taskRun.Status.Status).To(Equal(kubechain.TaskRunStatusStatusError))
+			// Phase shouldn't be Failed for general errors
+			Expect(taskRun.Status.Phase).ToNot(Equal(kubechain.TaskRunPhaseFailed))
```

Comment:
this `ToNot(Equal` was a little confusing. My eyes skipped the `Not` - why not just assert what the actual phase should be?